tests/test_ancillaries.py

```diff
@@ -72,6 +124,76 @@ def test_af_deform():
     assert np.allclose(ds["deform"], 1 - ds["circ"])
 
 
+def test_af_shared_af_method_pipeline_called_once_hdf5():
+    """Uses `setup_cleanup_fake_ancillary_features` as a fixture"""
+    path = "rtdc_data_hdf5_rtfdc.zip"
+    feature_name1, feature_name2 = "userdef1", "userdef2"
+    ds, af1, af2 = setup_fake_af(feature_name1, feature_name2, path)
+
+    assert example_shared_af_method.calls == 0
+    assert len(ds._ancillaries) == 0
+
+    _ = ds[feature_name1]
+    assert example_shared_af_method.calls == 1
+    assert len(ds._ancillaries) == 2
+
+    _ = ds[feature_name2]
+    assert example_shared_af_method.calls == 1, "method is not called again"
+    assert len(ds._ancillaries) == 2
+    feats = [feature_name1, feature_name2]
```

Comment:
these lines should probably be above the `ds[feature_name2]` call to show that they are populated by `ds[feature_name1]`
Same for line 168

Reply:
Yes, I agree. You could also remove the for-loop to reduce complexity.